### ISSUE TITLE
Crash on exit in TTS_Linux destructor in IDE

### ISSUE THREAD

**Issue by lawnjelly** [godotengine/godot#72844]
### Godot version

4.0 beta 17

### System information

Linux Mint 20.1, Intel i5-7500T, Intel HD graphics 630, 8gb

### Issue description

The IDE catches a crash on exit, call stack:
```
1 __GI___libc_read       read.c           26  0x7ffff7d4600c 
2 __GI___libc_read       read.c           24  0x7ffff7d4600c 
3 _IO_new_file_underflow libioP.h         948 0x7ffff7cc8b9f 
4 _IO_getdelim           iogetdelim.c     73  0x7ffff7cbb718 
5 ??                                          0x7ffff4376b39 
6 ??                                          0x7ffff4376d48 
7 start_thread           pthread_create.c 477 0x7ffff7f87609 
8 clone                  clone.S          95  0x7ffff7d57133 
```
From debug stepping, it occurs in `tts_linux.cpp`, in the destructor:
```
	if (synth) {
		spd_close(synth); <<<<< HERE
	}
```

### Steps to reproduce

Run project manager on the above system.
Note: Crash doesn't seem to occur or be reported in console outside the IDE.

### Minimal reproduction project

N/A.

### Discussion
@akien-mga suggested it is likely related to #71263 . Could be a mismatch of headers with an installed version?

**Comment by bruvzg** [comment]
Can't reproduce, might be a bug in a specific libspeechd version? https://github.com/brailcom/speechd/issues/762

**Comment by akien-mga** [comment]
Couldn't find proper packages listing for Linux Mint 20.1 but AFAIK it's based of Ubuntu 20.04, which indeed has libspeechd 0.9.1: https://packages.ubuntu.com/focal/libspeechd2

The upstream bug linked above is fixed in 0.11.3 (latest is 0.11.4).

I doubt Ubuntu 20.04 and derivatives will update the lib, we can maybe get some of them to backport the patch, but if there's a way for us to work it around, that might be best.

That being said, the linked bug seems to be a regression fix for a bug introduced in 0.11.2 so it's probably not what crashes on Linux Mint 20.1.

**Comment by lawnjelly** [comment]
It starts occurring on 4.0 beta 15. Not present in beta 14.

**Comment by bruvzg** [comment]
> It starts occurring on 4.0 beta 15. Not present in beta 14.

Beta 14 and older were build without TTS enabled, since the build environment did not have dev packages installed, https://github.com/godotengine/godot/pull/71263 allowed it to build without dev packages.

**Comment by lawnjelly** [comment]
Given that it only seems to get caught in the IDE (QTCreator), and if it is an external bug then I guess ignoring it is an option. Maybe there's a way I can turn off TTS in the build / runtime, or just ignore the crash on exit. Really raised this in case it was a bug in our code.

**Comment by bruvzg** [comment]
>  Maybe there's a way I can turn off TTS in the build / runtime

`speechd=false` in the scons args.

**Comment by akien-mga** [comment]
Fixed by #73671 I assume.
